resteasy-core-spi/src/main/java/org/jboss/resteasy/concurrent/ContextualExecutors.java

```diff
@@ -323,20 +322,13 @@ private static Runnable runnable(final Map<ThreadContext<Object>, Object> contex
     @SuppressWarnings("unchecked")
     private static Map<ThreadContext<Object>, Object> getContexts() {
         final Map<ThreadContext<Object>, Object> contexts = new LinkedHashMap<>();
-        if (System.getSecurityManager() == null) {
-            ServiceLoader.load(ThreadContext.class).forEach(context -> contexts.put(context, context.capture()));
-        } else {
-            AccessController.doPrivileged((PrivilegedAction<Object>) () -> {
-                ServiceLoader.load(ThreadContext.class).forEach(context -> contexts.put(context, context.capture()));
-                return null;
-            });
-        }
         // Load any registered providers
-        final ThreadContexts threadContexts = ResteasyProviderFactory.getInstance().getContextData(ThreadContexts.class);
-        if (threadContexts != null) {
-            for (ThreadContext<Object> context : threadContexts.getThreadContexts()) {
-                contexts.put(context, context.capture());
-            }
+        ThreadContexts threadContexts = ResteasyProviderFactory.getInstance().getContextData(ThreadContexts.class);
+        if (threadContexts == null) {
+            threadContexts = new ThreadContexts();
```

Comment:
Hi @jamezp .  Did the customer say this fixed the issue?  For us, we still see it going through the serviceloader for each request that flows through this.   `threadContexts` always appears to be `null`.  What am I missing?

Reply:
@crankydillo No, it didn't end up fixing it, https://issues.redhat.com/browse/RESTEASY-3313. It cut down on it, but didn't fix it. I'm going to have to see if there is a better place to put it.

I'm currently trying to see if it's reasonable to rewrite the entire context and `ResteasyProviderFactory` stuff now as I think there are better ways to handle it.

Reply:
As always, thanks for the fast-feedback!  Apologies for not searching JIRA.  I should have done that!

Reply:
No worries. I was looking at it yesterday so I had the JIRA up already in a tab :)